### ISSUE TITLE
Cannot parse `create domain` (postgres)

### ISSUE THREAD

**Issue by NBprojekt** [sqlfluff/sqlfluff#3256]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The postgres feature [create domain](https://www.postgresql.org/docs/current/sql-createdomain.html) can not be parsed.  

```
== [src\02_domains.sql] FAIL
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'create
                       | domain oname as text'
```

The same applies for `alter domain`.  
Interesting is that `comment on domain` does work.

### Expected Behaviour

Should be parsed?  ¯\_(ツ)_/¯

### Observed Behaviour

```
$ sqlfluff lint src/test.sql
== [src\test.sql] FAIL
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'create
                       | domain oname as text'
L:   2 | P:   1 |  PRS | Line 2, Position 1: Found unparsable section: 'alter
                       | domain oname add constraint "test"...'
L:   2 | P:  64 | L039 | Unnecessary whitespace found.
All Finished 📜 🎉!
```

### How to reproduce

Create `test.sql` with the following content:
```sql
create domain oname as text;
alter domain oname add constraint "test" check (length(value) <  512);
```
and run `sqlfluff lint test.sql`
> the config is below

### Dialect

postgres

### Version

sqlfluff 0.13.0
Python 3.9.7

### Configuration

[sqlfluff]
dialect = postgres
sql_file_exts = .sql

[sqlfluff:rules]
tab_space_size = 2
max_line_length = 80
indent_unit = space
comma_style = trailing

[sqlfluff:rules:L057]  # Special characters in identifiers
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
allow_space_in_identifier = False
additional_allowed_characters = "-."

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
